Assets/Scripts/Algorithms/Patrolling/Components/CollisionRecoverySenseNeighboringRobotsComponent.cs

```diff
@@ -0,0 +1,125 @@
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+// Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License along
+// with MAEPS. If not, see http://www.gnu.org/licenses/.
+//
+// Contributors 2025: 
+// Casper Nyvang Sørensen,
+// Christian Ziegler Sejersen,
+// Jakob Meyer Olsen
+// Mads Beyer Mogensen
+using System.Collections.Generic;
+using System.Text;
+
+using Maes.Map.PathFinding;
+using Maes.Robot;
+using Maes.Robot.Tasks;
+
+namespace Maes.Algorithms.Patrolling.Components
+{
+    internal class CollisionMessage
+    {
+        public CollisionMessage(int myId, Queue<PathStep> currentPath)
+        {
+            RobotId = myId;
+            PathSteps = currentPath;
+        }
+
+        public int RobotId { get; }
+        public Queue<PathStep> PathSteps { get; }
+    }
+    public class CollisionRecoverySenseNeighboringRobotsComponent : IComponent
+    {
+        private const float RecoveryAngle = 45f; // All robots rotate by this angle
+        private readonly Robot2DController _controller;
+        private bool _doingCollisionRecovery;
+
+        public int PreUpdateOrder => -100;
+        public int PostUpdateOrder => -100;
+
+        public CollisionRecoverySenseNeighboringRobotsComponent(Robot2DController controller)
+        {
+            _controller = controller;
+        }
+        public IEnumerable<ComponentWaitForCondition> PreUpdateLogic()
+        {
+            while (true)
+            {
+                if (_controller.IsCurrentlyColliding)
+                {
+                    _doingCollisionRecovery = true;
+                    _controller.StopCurrentTask();
+                    yield return ComponentWaitForCondition.WaitForRobotStatus(RobotStatus.Idle, false);
+
+                    GetNearbyRobots(out var myId, out var lowestId);
+
+                    if (lowestId == myId)
+                    {
+                        foreach (var condition in PerformRecoveryManeuver())
+                        {
+                            yield return condition;
+                        }
+
+                        while (lowestId == myId)
+                        {
+                            yield return ComponentWaitForCondition.WaitForLogicTicks(1, false);
+                            GetNearbyRobots(out myId, out lowestId);
+                        }
+                    }
+                }
+
+                yield return ComponentWaitForCondition.WaitForLogicTicks(1, true);
+                _doingCollisionRecovery = false;
+            }
+        }
+
+        public void DebugInfo(StringBuilder stringBuilder)
+        {
+            if (_doingCollisionRecovery)
+            {
+                stringBuilder.Append("Doing multi-robot collision avoidance\n");
+            }
+        }
+
+        private IEnumerable<ComponentWaitForCondition> PerformRecoveryManeuver()
```

Comment:
What is this magic dance? Is it tested, cause it looks like it was taken from HenrikExplorationAlgorithm which was... not a success?